qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/engineering_diffraction_io.py

```diff
@@ -84,7 +84,7 @@ def decode(obj_dic, _=None):
             fit_data_widget = presenter.fitting_presenter.data_widget
             fit_data_widget.model.restore_files(ws_names)
             fit_data_widget.presenter.plotted = set(obj_dic["plotted_workspaces"])
-            fit_data_widget.presenter.restore_table()
+            fit_data_widget.presenter.restore_table(clear_plotted=False)
```

Comment:
It might be worth documenting why we have to call this with `clear_plotted = False` - something along the lines of (summarised from your slack message)

> In the test case there is no worker used (FittingDataPresenter.worker=None), which causes the `_handle_table_cell_changed` not to get called when the slot has `QtCore.Qt.QueuedConnection` in  `self.table_selection.cellChanged.connect(slot, QtCore.Qt.QueuedConnection)`

If that's accurate...

Reply:
Done